src/Microsoft.Kusto.ServiceLayer/DataSource/ReliableDataSourceConnection.cs

```diff
@@ -224,6 +215,34 @@ public Task OpenAsync(CancellationToken token)
             }
         }
 
+        public async Task<IDataReader> ExecuteQueryAsync(string query, CancellationToken cancellationToken, string databaseName)
+        {
+            return await _commandRetryPolicy.ExecuteAction(() =>
+                _dataSource.ExecuteQueryAsync(query, cancellationToken, databaseName));
+        }
+
+        public IEnumerable<DataSourceObjectMetadata> GetChildObjects(DataSourceObjectMetadata objectMetadata,
+            bool includeSizeDetails = false)
+        {
+            return _commandRetryPolicy.ExecuteAction(() =>
+                _dataSource.GetChildObjects(objectMetadata, includeSizeDetails));
+        }
+
+        public DiagnosticsInfo GetDiagnostics(DataSourceObjectMetadata objectMetadata)
+        {
+            return _commandRetryPolicy.ExecuteAction(() => _dataSource.GetDiagnostics(objectMetadata));
+        }
+
+        public string GenerateAlterFunctionScript(string functionName)
+        {
+            return _commandRetryPolicy.ExecuteAction(() => _dataSource.GenerateAlterFunctionScript(functionName));
+        }
+
+        public string GenerateExecuteFunctionScript(string functionName)
+        {
+            return _commandRetryPolicy.ExecuteAction(() => _dataSource.GenerateExecuteFunctionScript(functionName));
+        }
+
```

Comment:
Added the above functions to not expose the Datasource to the rest of the project.